### PR TITLE
feat(cb2-13861): added script for generating batch plate data

### PR DESCRIPTION
**Pull request by LGin-BJSS**
## Description

added a script for generating batch plate data

Related issue: [CB2-13861](https://dvsa.atlassian.net/browse/CB2-13861)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by naathanbrown** [comment]
DRAFT DON'T MERGE

File: tests/util/generateBatchPlateData.ts
Comment (Daniel-Searle) [inline tests/util/generateBatchPlateData.ts]: isn't this assuming that these files are already there? If they're not, will they throw an error?

Comment (matthew2564) [reply tests/util/generateBatchPlateData.ts]: @Daniel-Searle  - As long as the `tests/resources/*` directory exists i.e. the parent, it should create the file gracefully, if not already there.

File: tests/util/generateBatchPlateData.ts
Comment (Daniel-Searle) [inline tests/util/generateBatchPlateData.ts]: Missing the testable and skeleton records

File: tests/util/generateBatchPlateData.ts
Comment (Daniel-Searle) [inline tests/util/generateBatchPlateData.ts]: upon running the script, the first couple of records contain the following details

>   "systemNumber": "XYZEP5JYOMM00020",
>     "createdTimestamp": "2019-06-22T10:26:55.903Z",

Doing this is going to cause them to be rejected when trying to put them into any dynamo db due to the overlap between the keys on the table.

File: tests/util/generateBatchPlateData.ts
Comment (Daniel-Searle) [inline tests/util/generateBatchPlateData.ts]: a lot of the functions, don't use console logging but use the winston logger instead

File: tests/util/generateBatchPlateData.ts
Comment (LGin-BJSS) [reply tests/util/generateBatchPlateData.ts]: My understanding is that our current seed process only uses the `technical-records-v3.json` file. This script appends to that with the intention of being used through the exisiting seed process. If our seed pulls together multiple files, that should quite quickly solve the issues we were seeing with the size of the JSON.

File: tests/util/generateBatchPlateData.ts
Comment (LGin-BJSS) [reply tests/util/generateBatchPlateData.ts]: This is really a throwaway script to show how the test data was generated. These logs are purely for tracking progress when invoking the file through `ts-node` in the CLI.

**Review by naathanbrown — APPROVED** [review]

File: src/handler/uploadPlateSeedData.ts
Comment (Daniel-Searle) [inline src/handler/uploadPlateSeedData.ts:44]: Fix for the Eslint issues, so that we don't need to switch them off! Do we need the event in the parameter - it's not being used anywhere
```suggestion
export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
  logger.info('Upload plate seed end point called');

  const fileNames = generateBatchPlateData();
  await Promise.all(fileNames.map((fileName) => processFile(fileName)));

  return addHttpHeaders({
    statusCode: 200,
    body: 'Uploaded all plate seed data',
  });
};

async function processFile(fileName: string): Promise<void> {
  console.log(fileName);
  const seedData = JSON.parse(fs.readFileSync(fileName, 'utf8')) as Record<string, unknown>[];

  const formattedSeedData = seedData.map((record) => ({
    PutRequest: {
      Item: marshall(record, { removeUndefinedValues: true }),
    },
  }));

  const chunkSize = 25;
  const chunks = [];
  for (let i = 0; i < formattedSeedData.length; i += chunkSize) {
    const chunk = formattedSeedData.slice(i, i + chunkSize);
    chunks.push(chunk);
  }

  await Promise.all(chunks.map(insertBatchPlateSeedData));
}

```

File: webpack/webpack.production.js
Comment (Daniel-Searle) [inline webpack/webpack.production.js]:
```suggestion
 'GenerateAdrCertificateFunction', 'RemoveInvalidPrimaryVrms', 'BatchPlateCreation', 'MotUpdateVrm', 'UploadPlateSeedDataFunction'];
```

File: tests/resources/technical-records-v3-with-batch-plates.json
Comment (Daniel-Searle) [inline tests/resources/technical-records-v3-with-batch-plates.json:1]: Now that we're inserting them via the new lambda- do we still need these changes in the jsons?

File: tests/resources/trigger-data.json
Comment (Daniel-Searle) [inline tests/resources/trigger-data.json:1]: Appreciate this is being generated locally okay, but when this is running on the lambda - where is the trigger data going to be?

**Comment by Daniel-Searle** [comment]
There are also conflicts on this branch

File: tests/resources/trigger-data.json
Comment (cb-cs) [inline tests/resources/trigger-data.json]: Does it make a difference that these are not camelCase like the original batchPlatesData.json data? To avoid having to manipulate further down the line.

File: src/util/generateBatchPlateData.ts
Comment (cb-cs) [inline src/util/generateBatchPlateData.ts]: In regards to my comment on trigger-data.json, just wondering if there is a reason this is not systemNumber and createdTimestamp?

File: tests/resources/trigger-data.json
Comment (naathanbrown) [reply tests/resources/trigger-data.json:1]: into the unknowwnnnnn

File: tests/resources/technical-records-v3-with-batch-plates.json
Comment (naathanbrown) [reply tests/resources/technical-records-v3-with-batch-plates.json:1]: no, will delete them from git before I merge, ty

File: src/util/generateBatchPlateData.ts
Comment (Daniel-Searle) [reply src/util/generateBatchPlateData.ts]: It needs to be camel cased, otherwise it will break the batch load lambda

Comment (naathanbrown) [reply src/util/generateBatchPlateData.ts]: caught up with Leo, think it's a boo boo, good spot

File: src/handler/uploadPlateSeedData.ts
Comment (naathanbrown) [reply src/handler/uploadPlateSeedData.ts:44]: No question this is better, but I'm just not willing to change so late to SG for something we are going to delete ASAP! Nice work tho man

**Pull request merged**
